framework/src/loops/AllLocalDofIndicesThread.C

```diff
@@ -21,24 +24,50 @@
 #include LIBMESH_INCLUDE_UNORDERED_SET
 LIBMESH_DEFINE_HASH_POINTERS
 
-AllLocalDofIndicesThread::AllLocalDofIndicesThread(System & sys,
+AllLocalDofIndicesThread::AllLocalDofIndicesThread(FEProblemBase & problem,
+                                                   System & sys,
```

Comment:
If we pass in the problem, we do not need to pass in the system here.

Reply:
Again, been there, done that. 

Problem is calling the constructor of the ParallelObject. If I use  `feproblem -> get system from variable name -> get communicator from system` I get a seg fault which shouldn't occur but I couldn't figure out why/how it happened. Obviously, the same call works later in the body of the constructor so I'm really at a loss. To make it work, I readded system.

Reply:
Well this is concerning. Since sys should not be the problem, something else is corrupting the memory. 
Did you step in with a debugger (to confirm that problem.getSystem is returning something good) ?
Can you run this in valgrind?

Reply:
No and I won't try. This is as far as I go here. I found a way around to satisfy my immediate needs.